### PR TITLE
Add support for aten::roll

### PR DESCRIPTION
**Pull request by bszmelcz**
## Details:
- [x] add transformation for `aten::roll`
- [x] add layer test for `aten::roll` operator
## Ticket:
94305

### CODE REVIEW

File: src/frontends/pytorch/src/op/roll.cpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+OutputVector translate_roll(NodeContext& context) {
+    const auto data = context.get_input(0);
+    const auto shifts = context.get_input(1);
+    const auto axes = context.get_input(2);
+    // if axes was not set
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/roll.cpp]: Please fix codestyle

File: tests/layer_tests/pytorch_tests/test_unbind.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Copyright (C) 2018-2022 Intel Corporation
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_unbind.py]: unbind test seems as change from other PR

File: tests/layer_tests/pytorch_tests/test_roll.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    @pytest.mark.parametrize(("shifts", "dim"), [
+            [(2, 1), (0, 1)],
+            [1, 0],
+            [-1, 0],
+            [1, None],
+    ])
+    @pytest.mark.nightly
+    def test_roll(self, shifts, dim, ie_device, precision, ir_version):
+        if ie_device == "CPU":
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_roll.py]: Please remove CPU restriction. (fyi you can use `export TEST_DEVICE=CPU` for local testing)

File: src/frontends/pytorch/src/transforms/prim_list_unpack_replacer.cpp [hunk 4]
```diff
@@ -51,6 +51,24 @@ PrimListUnpackReplacer::PrimListUnpackReplacer() {
             return true;
         }
 
+        if (auto unbind = cast_fw_node(input_node, "aten::unbind")) {
```
Comment (mvafin) [inline src/frontends/pytorch/src/transforms/prim_list_unpack_replacer.cpp]: this also looks like change from other PR

File: src/frontends/pytorch/src/op/roll.cpp [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+OutputVector translate_roll(NodeContext& context) {
+    const auto data = context.get_input(0);
+    const auto shifts = context.get_input(1);
+    const auto axes = context.get_input(2);
+    // if axes was not set 
+    if (axes.get_shape() != shifts.get_shape()) {
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/roll.cpp]: if `axes` is not provided what `axes.get_shape()` returns?

Comment (bszmelcz) [reply src/frontends/pytorch/src/op/roll.cpp]: If it is not provided, default it 0

Comment (mvafin) [reply src/frontends/pytorch/src/op/roll.cpp]: By this verification you restrict `axes` and `shifts` to have static shape, but I don't think this is required.

Comment (bszmelcz) [reply src/frontends/pytorch/src/op/roll.cpp]: I can add a check that also tests if both axes and shifts have dynamic shapes. Would be that sufficient?

File: tests/layer_tests/pytorch_tests/test_roll.py (hunk 3 above)
Comment (bszmelcz) [reply tests/layer_tests/pytorch_tests/test_roll.py]: Thanks!

File: src/frontends/pytorch/src/op/roll.cpp [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+namespace op {
+
+OutputVector translate_roll(NodeContext& context) {
+    const auto data = context.get_input(0);
+    const auto shifts = context.get_input(1);
+    const auto axes = context.get_input(2);
+    const auto shifts_pshape = shifts.get_partial_shape();
+    const auto axes_pshape = axes.get_partial_shape();
+    const auto match_dims = axes_pshape.compatible(shifts_pshape);
+    const auto both_dynamic = (shifts_pshape.rank().is_dynamic() && axes_pshape.rank().is_dynamic());
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/roll.cpp]: `compatible` from line 20 also checks for dynamic shapes, you do not need to check it again.

Comment (bszmelcz) [reply src/frontends/pytorch/src/op/roll.cpp]: Got it!

**Review by mvafin — APPROVED** [review]

**Pull request merged**